### PR TITLE
fix(NewConversationDialog): speed up conversation creation flow

### LINKED ISSUE #12206: Participant list is only displayed after some time
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Start a new conversation
2. Invite no or one participant

### Expected behaviour
The participants list should show all participants immediately

### Actual behaviour
It takes some times 10 or moure (up to a feeled minute) seconds until the participants list are shown. Before that only grey placeholders are shown (see screenshot).

## Talk app

spreed: 18.0.7

**Custom Signaling server configured:** no

**Custom TURN server configured:** yes

**Custom STUN server configured:** no

## Browser

**Microphone available:** error doesn't depend on this

**Camera available:** error doesn't depend on this

**Operating system:** error doesn't depend on this

**Browser name:** Error occurs in all tested browsers

**Browser version:** 

### Browser log

```
Content-Security-Policy: Ignorieren von "blob:" innerhalb script-src-elem: 'strict-dynamic' angegeben [xd7qauji](https://cloud.example.org/call/xd7qauji)
Die Ressource von "https://cloud.example.org/apps/tasks/css/tasks-talk.css?v=b5ecd63e-37" wurde wegen eines MIME-Typ-Konfliktes ("text/html") blockiert (X-Content-Type-Options: nosniff).
[xd7qauji](https://cloud.example.org/call/xd7qauji)
Proxying an event bus of version 3.1.0 with 1.3.0 [index.es.js:2337:14](webpack://sualko/cloud_bbb/node_modules/@nextcloud/event-bus/dist/index.es.js)
[WARN] viewer: Some mimes were ignored because they are not enabled in the server previews config 
Object { app: "viewer", uid: "xyadmin", level: 2, ignoredMimes: (3) […] }
[ConsoleLogger.js:71:18](webpack:///viewer/node_modules/@nextcloud/logger/dist/ConsoleLogger.js)
Diese Seite verwendet die nicht standardisierte Eigenschaft "zoom". Stattdessen sollte calc() in den entsprechenden Eigenschaftswerten oder "transform" zusammen mit "transform-origin: 0 0" verwendet werden. [xd7qauji](https://example.org/call/xd7qauji)
Message to focus not found in DOM 18457 [MessagesList.vue:958:12](webpack:///talk/src/components/MessagesList/MessagesList.vue)
OCA.Sharing.ShareSearch not ready [init-collections.js:39:2](webpack:///deck/src/init-collections.js)
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
```

</details>

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Nginx

**Database:** MySQL / Galera

**PHP version:** 8.1

**Nextcloud Version:** 28.0.4.1

**List of activated apps:**

<details>

```
Enabled:
  - activity: 2.20.0
  - announcementcenter: 6.8.1
  - appointments: 2.0.7
  - bbb: 2.5.0
  - bruteforcesettings: 2.8.0
  - calendar: 4.7.0
  - circles: 28.0.0
  - cloud_federation_api: 1.11.0
  - collectives: 2.10.1
  - comments: 1.18.0
  - contacts: 5.5.3
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - deck: 1.12.2
  - external: 5.3.1
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - forms: 4.2.3
  - groupfolders: 16.0.6
  - impersonate: 1.15.0
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - mail: 3.6.0
  - nextcloud_announcements: 1.17.0
  - notes: 4.9.4
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - password_policy: 1.18.0
  - photos: 2.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - richdocuments: 8.3.4
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - side_menu: 3.12.0
  - spreed: 18.0.7
  - support: 1.11.1
  - survey_client: 1.16.0
  - systemtags: 1.18.0
  - tasks: 0.15.0
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
Disabled:
  - admin_audit: 1.18.0
  - encryption: 2.16.0
  - files_external: 1.20.0 (installed 1.13.0)
  - files_mindmap: 0.0.30 (installed 0.0.29)
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - suspicious_login: 6.0.0
  - twofactor_totp: 10.0.0-beta.2
  - user_ldap: 1.19.0

```
</details>

**Nextcloud configuration:**

<details>

```
{
    "system": {
        "check_for_working_wellknown_setup": false,
        "default_language": "de",
        "default_locale": "de_DE",
        "default_phone_region": "DE",
        "log_rotate_size": 104857600,
        "logdateformat": "Y-m-d H:i:s.u",
        "loglevel": 2,
        "logtimezone": "Europe\/Berlin",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "preview_max_filesize_image": 2,
        "preview_max_x": 1280,
        "preview_max_y": 1280,
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": "0.0"
        },
        "skeletondirectory": "",
        "trusted_domains": [
            "cloud.example.org"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "updatechecker": true,
        "updater.release.channel": "stable",
        "upgrade.disable-web": true,
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "28.0.4.1",
        "overwrite.cli.url": "https:cloud.example.org",
        "overwriteprotociol": "https",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": true,
        "mail_smtpauthtype": "PLAIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpsecure": "ssl",
        "maintenance_window_start": 1,
        "app_install_overwrite": [
            "bbb"
        ]
    }
}
```
</details>

### Server log (data/nextcloud.log)
<details>

```
# no errors
```
</details>
Auswahl_023

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #12206 
* Only `createConversation` is mandatory to be completed first, rest request are secondary and could be done in parallel
* As owner of conversation definitely has all needed permissions and access to the conversation, we can fetch participants list (with statuses, permissions, e.t.c) much earlier
  * we don't need to wait to join conversation
  * we're the only one in the conversation yet, so don't need to debounce request (up to 3-15 sec)

### 🖼️ Screenshots / Screencasts

Testing the time (with local server) from **submitting the form** with following:
- Description
- Avatar
- Password
- Listable
- 5 participants

to the time **participants are fetched** 

🏚️ Before | 🏡 After
-- | --
image | image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/ConversationSettings/ConversationAvatarEditor.vue [hunk 1]
```diff
@@ -320,11 +320,11 @@ export default {
 			}
 		},
 
-		savePictureAvatar() {
+		async savePictureAvatar() {
 			this.showCropper = false
 			const canvasData = this.$refs.cropper.getCroppedCanvas()
 			const scaleFactor = canvasData.width > 512 ? 512 / canvasData.width : 1
-			this.$refs.cropper.scale(scaleFactor, scaleFactor).getCroppedCanvas().toBlob(async (blob) => {
+			await this.$refs.cropper.scale(scaleFactor, scaleFactor).getCroppedCanvas().toBlob(async (blob) => {
```
Comment (ShGKme) [inline src/components/ConversationSettings/ConversationAvatarEditor.vue:327]: `toBlob` does not return any `promise`
https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob

**Review by ShGKme — APPROVED** [review]

**Comment by ShGKme** [comment]
cc @nickvergessen 
It seems we need too many requests to create a conversation. Can we add a new entrypoint / extend existing to do everything in one/two requests instead `5 + n` where N is the number of participants?

**Comment by Antreesy** [comment]
> Can we add a new entrypoint / extend existing

Previous discussion in https://github.com/nextcloud/spreed/issues/2874

**Pull request merged**